pkg/cache/v3/resource.go

```diff
@@ -215,14 +215,28 @@ func getListenerReferences(src *listener.Listener, out map[resource.Type]map[str
 			routeSpecifier := config.RouteSpecifier
 			switch r := routeSpecifier.(type) {
 			case *hcm.HttpConnectionManager_Rds:
-				if r != nil && r.Rds != nil {
-					routes[r.Rds.RouteConfigName] = true
+				if r == nil || r.Rds == nil {
+					break
```

Comment:
I don't understand the break here. This will cause the function to return an incomplete list of resource references. Shouldn't it be a continue?

Reply:
This breaks from the innermost switch scope

Reply:
Right.